keras_nlp/models/t5/t5_backbone.py

```diff
@@ -0,0 +1,846 @@
+            output_dim=output_dim,
+            embeddings_initializer=keras.initializers.TruncatedNormal(1.0),
+            name="shared_embedding",
+        )
+        encoder = T5MainLayer(
+            is_decoder=False,
+            use_gated_activation=use_gated_activation,
+            output_dim=output_dim,
+            hidden_dim=hidden_dim,
+            dropout=dropout,
+            activation=activation,
+            layer_norm_epsilon=layer_norm_epsilon,
+            key_value_dim=key_value_dim,
+            num_heads=num_heads,
+            num_blocks=num_blocks,
+            token_embeddings=shared_embedding,
+            name="encoder",
+        )
+        decoder = T5MainLayer(
+            is_decoder=True,
+            use_gated_activation=use_gated_activation,
+            output_dim=output_dim,
+            hidden_dim=hidden_dim,
+            dropout=dropout,
+            activation=activation,
+            layer_norm_epsilon=layer_norm_epsilon,
+            key_value_dim=key_value_dim,
+            num_heads=num_heads,
+            num_blocks=num_blocks,
+            token_embeddings=shared_embedding,
+            name="decoder",
+        )
+        token_ids = keras.Input(shape=(None,), dtype="int32", name="token_ids")
+        decoder_token_ids = keras.Input(
+            shape=(None,), dtype="int32", name="decoder_token_ids"
+        )
+        encoder_outputs = encoder(token_ids)
+        decoder_outputs = decoder(
+            decoder_token_ids,
+            encoder_hidden_states=encoder_outputs,
+        )
+        super().__init__(
+            {"token_ids": token_ids, "decoder_token_ids": decoder_token_ids},
+            decoder_outputs,
+            **kwargs,
+        )
+        # All references to `self` below this line
+        self.vocabulary_size = vocabulary_size
+        self.output_dim = output_dim
+        self.hidden_dim = hidden_dim
+        self.key_value_dim = key_value_dim
+        self.num_blocks = num_blocks
+        self.num_heads = num_heads
+        self.use_gated_activation = use_gated_activation
+        self.activation = activation
+        self.dropout = dropout
+        self.layer_norm_epsilon = layer_norm_epsilon
+
+    def get_config(self):
+        config = super().get_config()
+        config.update(
+            {
+                "vocabulary_size": self.vocabulary_size,
+                "output_dim": self.output_dim,
+                "hidden_dim": self.hidden_dim,
+                "key_value_dim": self.key_value_dim,
+                "num_blocks": self.num_blocks,
+                "num_heads": self.num_heads,
+                "use_gated_activation": self.use_gated_activation,
+                "activation": self.activation,
+                "dropout": self.dropout,
+                "layer_norm_epsilon": self.layer_norm_epsilon,
+            }
+        )
+        return config
+
+    @classproperty
+    def presets(cls):
+        return copy.deepcopy(backbone_presets)
```

Comment:
We have started directly exposing the `token_embedding` as a property as it is pretty common to need to use it directly. https://github.com/keras-team/keras-nlp/blob/master/keras_nlp/models/bart/bart_backbone.py#L255-L257

Reply:
Done.